### PR TITLE
v3.1.0 changes.

### PR DESCRIPTION
**Pull request by fcd1**

### CODE REVIEW

File: app/controllers/deposits_controller.rb [hunk 1]
```diff
@@ -63,6 +63,58 @@ def destroy
...
+    Rails.logger.warn("Following is a re-deposit:" \
+                      "Title: #{endpoint.deposit_title.truncate_words(10)}, " \
+                      "Files: #{endpoint.documents_to_deposit}, " \
+                      "Hyacinth item pid: #{endpoint.adapter_item_identifier}, " \
+                      "Hyacinth asset pids: #{endpoint.asset_pids}, " \
+                      "Path to SWORD contents: #{path_to_deposit_contents}"
+                     )
+
+    # create Deposit instance to store deposit info in database
+    resubmit_deposit = Deposit.new
```
Comment (JackBlackLight) [inline app/controllers/deposits_controller.rb:106]: in the interest of keeping the controller slim, this deposit creation logic could be moved to the deposit model

**Comment by JackBlackLight** [comment]
@fcd1, this looks good to me - i just wanted your feeback on a couple of questions and then i can close this out. thanks.

File: app/controllers/deposits_controller.rb [hunk 2]
```diff
@@ -63,6 +63,58 @@ def destroy
...
+
+  def resubmit
+    original_deposit = Deposit.find(params[:id])
+    endpoint = get_endpoint(original_deposit.collection_slug,
+                            original_deposit.depositor_user_id)
+
+    path_to_deposit_contents = original_deposit.content_path
+
+    # log basic essential info. Keep it terse! Gonna use :warn level, though not a warning.
+    Rails.logger.warn("About to resubmit deposit for deposit id: #{original_deposit.id}:" \
```
Comment (JackBlackLight) [inline app/controllers/deposits_controller.rb:88]: just wanted to mention how much i appreciate your approach to logging in this app. the thorough logs have been a huge help in troubleshooting self-deposit issues in the past. thanks!

**Review by JackBlackLight — APPROVED** [review]
@fcd1, I'm only in for a half day tomorrow and then I'm out until next Wednesday so I'm going to approve this PR so as to not slow things down.

File: app/controllers/deposits_controller.rb [hunk 3]
```diff
@@ -63,6 +63,58 @@ def destroy
...
+                   depositor_user_id)
+    case COLLECTIONS[:slug][collection_slug][:parser]
+    when "academic-commons"
+      Sword::Endpoints::AcademicCommonsEndpoint.new(collection_slug, depositor_user_id)
+    when "proquest"
+      Sword::Endpoints::ProquestEndpoint.new(collection_slug, depositor_user_id)
+    when "eprints"
+      Sword::Endpoints::EprintsEndpoint.new(collection_slug, depositor_user_id)
+    else
+      # raise an exception here
```
[the remark this answers is not in the document]
Comment (fcd1) [reply app/controllers/deposits_controller.rb:76]: I created SWORD-118 to be used to add the exception in a later version.

File: config/credentials.yml.enc [hunk 4]
```diff
@@ -0,0 +1 @@
+v06g4L5370ws2l84uAwwWjVzFth5pYEZ/EJjI3/q71d9zYkq1JrHuCAAZh1fNzvbEd3m13n08m/CK+whQNBFSntoMKtTQ4gI6l4K3xWPLULR2UKZnIR7kQ0rTkO7PHtoMHvIS4GkEwkelbofvYoSwMuJElrKwZvtDB1c3GZotal5RYpMrvcaAd9/zCp7/6YvG5w/JDONe5sfhC8b3TWEaoAOagDOOxFv4kyn33uGKBJ5mqRB39nbp2SE8tkp9U2pORFFmlqGv96QyIh7dT0uzxCsY6K7AuZU6cmZ9wEuRmt7B0ejoL2iW6lk4hYAaP14mwif7VI0R+MM7MtnBZxOcALpwqvTaSuX1fmTbLxC2V79KCRbPY+ws3e8s6nW7FdHRR6HrRvBFrlttHhxLrCDRmVLwlN4--yc+eua1j1+T3zHL3--00tSaNKnHicurdZXqRzuRQ==
```
[the remark this answers is not in the document]
Comment (fcd1) [reply config/credentials.yml.enc:1]: With the new way to store credentials (https://guides.rubyonrails.org/v7.2/security.html#custom-credentials), credentials.yml.enc is encrypted and can therefore be stored in version control.

File: app/controllers/deposits_controller.rb (hunk 1 above)
Comment (fcd1) [reply app/controllers/deposits_controller.rb:106]: I did consider that option. I've opened a ticket, SWORD-119, and I will investigate (and possibly implement) that option.

**Pull request merged**